### PR TITLE
Add listKnownParties to DAML Script

### PR DESCRIPTION
**Pull request by cocreature**
This is particularly useful if you want easy access to a party that
has already been allocated since partyFromText is bad.

For now this is not supported in the JSON API. It should be possible
to add it but I consider it fairly low priority so omitting for now.

changelog_begin

- [DAML Script] Add ``listKnownParties`` and ``listKnownPartiesOn`` to
  query the corresponding ListKnownParties endpoint in the party
  management service.

changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-script/test/daml/ScriptTest.daml [hunk 1]
```diff
@@ -171,6 +172,15 @@ partyIdHintTest = do
   dan <- allocatePartyWithHint "dan" (PartyIdHint "dan")
   pure (carol, dan)
 
+listKnownPartiesTest : Script ([PartyDetails], Party)
+listKnownPartiesTest = do
+  before <- listKnownParties
+  p <- allocateParty "myparty"
```
Comment (hurryabit) [inline daml-script/test/daml/ScriptTest.daml:178]: Maybe a more specific name like `ListKnownPartiesTestParty` is less prone to accidental collisions with future tests?

File: daml-script/test/daml/MultiTest.daml [hunk 2]
```diff
@@ -39,3 +40,13 @@ partyIdHintTest = do
...
   pure (alice, bob)
+
+listKnownPartiesTest : Script ([PartyDetails], [PartyDetails])
+listKnownPartiesTest = do
+  parties1 <- listKnownPartiesOn (ParticipantName "one")
+  parties2 <- listKnownPartiesOn (ParticipantName "two")
+  p1 <- allocatePartyOn "p1" (ParticipantName "one")
+  p2 <- allocatePartyOn "p2" (ParticipantName "two")
+  parties1' <- listKnownPartiesOn (ParticipantName "one")
+  parties2' <- listKnownPartiesOn (ParticipantName "two")
```
Comment (hurryabit) [inline daml-script/test/daml/MultiTest.daml:51]: Shouldn't we have some assertions here as well? And do I get it right that we can't make assertions on the `isLocal` field here? I assume that doesn't stop us from asserting on the other two fields?

File: daml-script/test/daml/ScriptTest.daml (hunk 1 above)
Comment (cocreature) [reply daml-script/test/daml/ScriptTest.daml:178]: This is the display name not the party id. Collisions shouldn’t matter here.

File: daml-script/test/daml/MultiTest.daml (hunk 2 above)
Comment (cocreature) [reply daml-script/test/daml/MultiTest.daml:51]: We could but then we’re just verifying the same thing we already verify for the single participant tests which seems a bit pointless. `is_local` would be the interesting part to check that stuff is actually from different participants but given this issue there isn’t much that we can test herre.

File: daml-script/test/daml/ScriptTest.daml (hunk 1 above)
Comment (hurryabit) [reply daml-script/test/daml/ScriptTest.daml:178]: Oh, right. Sleepy me... Pardon!

File: daml-script/test/daml/MultiTest.daml (hunk 2 above)
Comment (hurryabit) [reply daml-script/test/daml/MultiTest.daml:51]: That makes sense. Thanks for the explanation.

**Pull request merged**